cpp/CMakeLists.txt

```diff
@@ -204,6 +204,9 @@ add_library(cugraph SHARED
     src/components/weakly_connected_components.cu
     src/structure/create_graph_from_edgelist.cu
     src/utilities/host_barrier.cpp
+    src/visitors/graph_envelope.cpp
```

Comment:
Something to think about: Where do we want this code to reside?

We've talked about making this a separate C-style API library.  It feels like that should have its own high level directory structure.  Perhaps something like: `src/api/visitors` and `include/cugraph/api`?  Just suggestions, I'm not sure what would work best.  But I think organizing the code so that all of the code for a particular library is grouped together will make the most sense.

Reply:
Done (most of it + see latest discussions / decisions on C-API)